docs/rules/no-get-with-default.md

```diff
@@ -0,0 +1,38 @@
+# No getWithDefault (no-get-with-default)
+
+This rule attempts to catch and prevent the use of `getWithDefault`.
+
+## Rule Details
+
+Even though the behavior for `getWithDefault` is more defined such that it only falls back to the default value on `undefined`,
+its inconsistency with the native `||` is confusing to many developers who assume otherwise. This rule encourages developers to use
+the native `||` operator instead.
```

Comment:
Perhaps we should mention the upcoming `??` operator as well? (as mentioned in the RFC)

Reply:
Done